test/functional/tests/io_class/test_io_class_lvm_on_cas.py

```diff
@@ -0,0 +1,153 @@
+
+    with TestRun.step("Run fio for file size from IO class from 11 to 21 "
+                      "and check that correct statistics increase."):
+        csv = casadm.list_io_classes(cache.cache_id, OutputFormat.csv).stdout
+        io_classes = IoClass.csv_to_list(csv)
+        file_size = Size(2, Unit.KibiByte)
+
+        for io_class in io_classes:
+            if io_class.id < 11 or io_class.id > 21:
+                continue
+
+            TestRun.LOGGER.info(f"IO Class ID: {io_class.id}, class name: {io_class.rule}")
+            cache.reset_counters()
+
+            TestRun.LOGGER.info(f"Running FIO with verification on LVM "
+                                f"[IO class ID {io_class.id}].")
+            run_fio(file_size, io_target)
+
+            TestRun.LOGGER.info(f"Checking statistics [IO class ID {io_class.id}].")
+            check_statistics(cache, io_classes, io_class.id)
+
+            if file_size < Size(256, Unit.MebiByte):
+                file_size *= 4
+            else:
+                file_size = 1100 * Size(1, Unit.MebiByte)
+
+            fs_utils.remove(io_target)
+
+    with TestRun.step("Remove LVMs."):
+        TestRun.executor.run(f"umount {mountpoint}")
+        Lvm.remove_all()
+
+
+def check_statistics(cache, io_classes: [], tested_class: int):
+    write_policy = cache.get_cache_mode()
+    stat_to_check = "Total requests"
+
+    for io_class in io_classes:
+        class_stats = cache.get_io_class_statistics(io_class.id)
+
+        stat_value = class_stats.request_stats.requests_total
+
+        if "metadata" in io_class.rule or "misc" in io_class.rule:
+            TestRun.LOGGER.info("Check skipped for tested IO Class for metadata and misc class")
+
+            TestRun.LOGGER.info(f"[{write_policy}] {stat_to_check} for IO Class {io_class.id} "
+                                f"[{io_class.rule}]: {stat_value}")
+            continue
+
+        if io_class.id == tested_class:
+            if stat_value == 0:
+                TestRun.LOGGER.error(f"[{write_policy}] {stat_to_check} equals 0 (but should "
+                                     f"increased) for IO Class {io_class.id} [{io_class.rule}]: "
+                                     f"{stat_value}")
+                TestRun.executor.run(f"ls -la {mountpoint}*")
+            else:
+                TestRun.LOGGER.info(f"[{write_policy}] {stat_to_check} for IO Class {io_class.id} "
+                                    f"[{io_class.rule}]: {stat_value}")
+                continue
+
+        if stat_value > 0:
+            TestRun.LOGGER.error(f"[{write_policy}] {stat_to_check} increased (not expected)"
+                                 f"for IO Class {io_class.id} [{io_class.rule}]: {stat_value}")
+            TestRun.executor.run(f"ls -la {mountpoint}*")
+        else:
+            TestRun.LOGGER.info(f"[{write_policy}] {stat_to_check} for IO Class {io_class.id} "
+                                f"[{io_class.rule}]: {stat_value}")
```

Comment:
Why move it to separate function? It's called only once.

Reply:
It was previously in "common" file. I've changed it, good point :)